nri-prelude/src/Task.hs

```diff
@@ -198,6 +200,23 @@ parallel tasks =
           |> Shortcut.map Prelude.sequence
     )
 
+-- | Given a task and a callback, execute the task in a greenthread
+-- and sends its result to callback.
+-- Always returns @OK ()@
+background :: forall x a . Task x a -> (Result x a -> Task x a) -> Task x ()
+background task callback =
+  Internal.Task
+    ( \handler ->
+        let runTask :: Task x a -> IO (Result x a)
+            runTask task' = Internal._run task' handler
+            waitForTask :: Async.Async (Result x a) -> IO (Result x a)
+            waitForTask greenTask = do
+              result <- Async.wait greenTask
+              runTask <| callback result
+         in Async.withAsync (runTask task) waitForTask
```

Comment:
i believe `withAsync` blocks until the action is done, e.g.

```haskell
main = do
  result <-
    withAsync (threadDelay 5_000_000) $ \promise ->
      Async.wait promise
      pure 4
  print result
```

i believe it will wait 5s before printing `4`.  we may want to use the lower-level `async` instead

Reply:
Indeed, but I think that happens due to the `result` dependency. It was my expectation that the usage of `*>` would avoid that block. Lemme check it :loading:

Reply:
That is indeed the case 😮
Changed in d7944f648ef59d618872e05b0146013b2e25307f